src/Compilers/Core/Portable/InternalUtilities/StringTable.cs

```diff
@@ -63,7 +63,11 @@ private struct Entry
         // slightly slower than local cache
         // we read this cache when having a miss in local cache
         // writes to local cache will update shared cache as well.
+#if COMPILERCORE
```

Comment:
>COMPILERCORE [](http://example.com/codeflow?start=4&length=12)

What does this mean? Who sets it an where? Consider adding this information as a comment

Reply:
It is at here: 

https://github.com/dotnet/roslyn/blob/fa7ece1ca25ef529db616ed6c839af9f027fa216/src/Compilers/Core/Portable/Microsoft.CodeAnalysis.csproj#L9

It allows us to share (link) code between IDE, compiler, and other layers, while customizing the code per-layer.  

As mentioned, I wrote it this way out of an abundance of caution. But I'll defer to compiler team if they would like no code difference, or if they would like this array out of the LOH as well. Thanks :-)

Reply:
> Consider adding this information as a comment 

Sure. I'll add a comment to that effect.

Reply:
Is there a reason for only doing it in the IDE though? Guessing the compiler would benefit from it as well.

Reply:
Did a [replay](https://github.com/dotnet/roslyn/tree/main/src/Tools/Replay) run on this change with the compiler and I see the same impact: 1MB less LOH allocations. Don't see a meaningful difference in throughput. Think we should just take this change for IDE + Compiler

Reply:
> Is there a reason for only doing it in the IDE though? 

paranoia :)

> Did a [replay](https://github.com/dotnet/roslyn/tree/main/src/Tools/Replay?rgh-link-date=2024-05-02T15%3A36%3A59Z) run on this change with the compiler and I see the same impact: 1MB less LOH allocations. Don't see a meaningful difference in throughput. Think we should just take this change for IDE + Compiler

Perfect!  Done :)